config.go

```diff
@@ -263,7 +255,7 @@ func loadConfig() (*config, []string, error) {
 	// Pre-parse the command line options to see if an alternative config
 	// file or the version flag was specified.
 	preCfg := cfg
```

Comment:
This has changed semantics now because cfg is a pointer, not a `config`, so it doesn't make a shallow copy of the struct.  Is this intentional?